controllers/storagecluster/generate.go

```diff
@@ -83,3 +85,31 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 func generateStorageQuotaName(storageClassName, quotaName string) string {
 	return fmt.Sprintf("ocs-%s-%s", storageClassName, quotaName)
 }
+
+// generateCephFSProviderParameters function generates extra parameters required for provider storage clusters
+func generateCephFSProviderParameters(initData *ocsv1.StorageCluster) (map[string]string, error) {
+	if !initData.Spec.AllowRemoteStorageConsumers {
+		return nil, nil
+	}
```

Comment:
redundant check, you are already checking on the call site. 
Assume that if this function is called then the config is needed